src/Mvc/Mvc.NewtonsoftJson/test/NewtonsoftJsonValidationMetadataProviderTest.cs

```diff
@@ -0,0 +1,89 @@
+        var propertyName = "sample-data";
+
+        var key = ModelMetadataIdentity.ForProperty(typeof(SampleTestClass).GetProperty(nameof(SampleTestClass.NoAttributesProperty)), typeof(int), typeof(SampleTestClass));
+        var modelAttributes = new ModelAttributes(Array.Empty<object>(), new[] { new JsonPropertyAttribute() { PropertyName = propertyName } }, Array.Empty<object>());
+        var context = new ValidationMetadataProviderContext(key, modelAttributes);
+
+        // Act
+        metadataProvider.CreateValidationMetadata(context);
+
+        // Assert
+        Assert.NotNull(context.ValidationMetadata.ValidationModelName);
+        Assert.Equal(propertyName, context.ValidationMetadata.ValidationModelName);
+    }
+
+    [Fact]
+    public void CreateValidationMetadata_SetValidationPropertyName_CamelCaseWithDefaultNamingPolicy()
+    {
+        var metadataProvider = new NewtonsoftJsonValidationMetadataProvider();
+        var propertyName = nameof(SampleTestClass.NoAttributesProperty);
+
+        var key = ModelMetadataIdentity.ForProperty(typeof(SampleTestClass).GetProperty(propertyName), typeof(int), typeof(SampleTestClass));
+        var modelAttributes = new ModelAttributes(Array.Empty<object>(), Array.Empty<object>(), Array.Empty<object>());
+        var context = new ValidationMetadataProviderContext(key, modelAttributes);
+
+        // Act
+        metadataProvider.CreateValidationMetadata(context);
+
+        // Assert
+        Assert.NotNull(context.ValidationMetadata.ValidationModelName);
+        Assert.Equal(new CamelCaseNamingStrategy().GetPropertyName(propertyName, false), context.ValidationMetadata.ValidationModelName);
+    }
+
+    [Theory]
+    [MemberData(nameof(NamingPolicies))]
+    public void CreateValidationMetadata_SetValidationPropertyName_WithJsonNamingPolicy(NamingStrategy namingStrategy)
+    {
+        var metadataProvider = new NewtonsoftJsonValidationMetadataProvider(namingStrategy);
+        var propertyName = nameof(SampleTestClass.NoAttributesProperty);
+
+        var key = ModelMetadataIdentity.ForProperty(typeof(SampleTestClass).GetProperty(propertyName), typeof(int), typeof(SampleTestClass));
+        var modelAttributes = new ModelAttributes(Array.Empty<object>(), Array.Empty<object>(), Array.Empty<object>());
+        var context = new ValidationMetadataProviderContext(key, modelAttributes);
+
+        // Act
+        metadataProvider.CreateValidationMetadata(context);
+
+        // Assert
+        Assert.NotNull(context.ValidationMetadata.ValidationModelName);
+        Assert.Equal(namingStrategy.GetPropertyName(propertyName, false), context.ValidationMetadata.ValidationModelName);
+    }
+
+    public static TheoryData<NamingStrategy> NamingPolicies
+    {
+        get
+        {
+            return new TheoryData<NamingStrategy>
+                {
+                    new UpperCaseJsonNamingPolicy(),
+                    new CamelCaseNamingStrategy()
+                };
```

Comment:
```suggestion
            {
                new UpperCaseJsonNamingPolicy(),
                new CamelCaseNamingStrategy()
            };
```

Nit, feel free to ignore it / do it in a separate PR